onlineguru/src/main/java/no/ntnu/online/onlineguru/plugin/plugins/flags/FlagsPlugin.java

```
@@ -103,8 +153,148 @@ private void handlePrivMsgEvent(PrivMsgEvent e) {
     }
 
     /*
+     * Internal helper methods
+     */
+
+    protected String serializeFlags(Set<Flag> flags) {
+        String result = "";
+        for (Flag f : flags) {
+            if (f == Flag.ANYONE) {
+                continue;
+            }
+            result += f;
+        }
+        return result;
+    }
+
+    protected Set<Flag> deserializeFlags(String flags) {
+        Set<Flag> result = new HashSet<Flag>();
+        if (flags != null) {
+            for (char c : flags.toCharArray()) {
+                try {
+                    Flag f = Flag.valueOf("" + c);
+                    result.add(f);
+                } catch (IllegalArgumentException e) {
+                    logger.error("Invalid flag recorded: " + c);
+                }
+            }
+        }
+        return result;
+    }
+
+    protected Set<Flag> updateFlags(Set<Flag> flags, String delta) {
+        char action = '.';
+        Flag currentFlag;
+
+        for (Character c : delta.toCharArray()) {
+            currentFlag = null;
+
+            switch (c) {
+                case '+':
+                case '-':
+                    action = c;
+                    break;
+                default:
+                    try {
+                        currentFlag = Flag.valueOf("" + c);
+                    } catch (IllegalArgumentException e) {
+                        logger.debug("Illegal flag '" + c + "'.");
```

It can fail silently, imo. The user will get a reply with the flags after the operation. ChanServ does this the same way.